### PR TITLE
split rest-api.js: split by api (except study api)

### PR DESCRIPTION
**Pull request by ghelloumi**
for this PR: 
- Create new folder named services
- Create one js file per api (except Study api: will be created in next sperated pull requests to avoid conlicts)

image

- Add getRequestParamFromList util to rest-api.js to uniform the urls creation

### CODE REVIEW

File: src/services/directory-notification.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+import ReconnectingWebSocket from 'reconnecting-websocket';
+import { getToken, getUrlWithToken } from '../utils/rest-api';
+
+const PREFIX_DIRECTORY_NOTIFICATION_WS = `${process.env.REACT_APP_WS_GATEWAY}/directory-notification`;
+
+export function connectDeletedStudyNotificationsWebsocket(studyUuid) {
+    // The websocket API doesn't allow relative urls
+    const wsbase = document.baseURI
+        .replace(/^http:\/\//, 'ws://')
+        .replace(/^https:\/\//, 'wss://');
```
Comment (Walid-Sahnoun) [inline src/services/directory-notification.js:17]: i think we can make it a function in utils

Comment (ghelloumi) [reply src/services/directory-notification.js:17]: rest-api is already in utils folder.
So I planned in next PRs for this TS to move all api calls functions to services folder and preserve only utils functions on rest-api.

thank you

Comment (ghelloumi) [reply src/services/directory-notification.js:17]: done, thank you

File: src/services/directory-notification.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+export function connectDeletedStudyNotificationsWebsocket(studyUuid) {
+    // The websocket API doesn't allow relative urls
+    const wsbase = getWsBase();
+    // Add params to Url
+    const urlSearchParams = new URLSearchParams();
+    urlSearchParams.append('updateType', 'deleteStudy');
+    urlSearchParams.append('elementUuid', studyUuid);
+
+    const wsadress = `${wsbase}${PREFIX_DIRECTORY_NOTIFICATION_WS}/notify?${urlSearchParams}`;
```
Comment (Walid-Sahnoun) [inline src/services/directory-notification.js:21]: rename to wsAdress

File: src/services/loadflow.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import { backendFetchJson } from '../utils/rest-api';
+
+const PREFIX_LOADFLOW_SERVER_QUERIES = `${process.env.REACT_APP_API_GATEWAY}/loadflow`;
+
+export function getLoadFlowUrl() {
+    return PREFIX_LOADFLOW_SERVER_QUERIES + '/v1/';
```
Comment (Walid-Sahnoun) [inline src/services/loadflow.js:13]: make '/v1' a const and exported

File: src/services/study-notification.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+import ReconnectingWebSocket from 'reconnecting-websocket';
+
+import { getUrlWithToken, getWsBase } from '../utils/rest-api';
+
+const PREFIX_STUDY_NOTIFICATION_WS = `${process.env.REACT_APP_WS_GATEWAY}/study-notification`;
+
+export function connectNotificationsWebsocket(studyUuid, options) {
+    // The websocket API doesn't allow relative urls
+    const wsbase = getWsBase();
+    const wsadress = `${wsbase}${PREFIX_STUDY_NOTIFICATION_WS}/notify?studyUuid=${encodeURIComponent(
```
Comment (Walid-Sahnoun) [inline src/services/study-notification.js:17]: rename to "wsBase" and "wsAdress"

File: src/services/loadflow.js (hunk 3 above)
Comment (ghelloumi) [reply src/services/loadflow.js]: Done, thank you
for v1 in rest-api, I will fix that later with next PRs.

File: src/services/study-notification.js (hunk 4 above)
Comment (ghelloumi) [reply src/services/study-notification.js:17]: Done, thank you

File: src/services/directory-notification.js (hunk 2 above)
Comment (ghelloumi) [reply src/services/directory-notification.js]: done, thank you

File: .env [hunk 5]
```diff
@@ -1,3 +1,4 @@
 REACT_APP_API_GATEWAY=api/gateway
 REACT_APP_WS_GATEWAY=ws/gateway
+REACT_APP_API_VERSION=v1
```
Comment (SlimaneAmar) [inline .env:3]: API_VERSION is not global but per service
Currently it's the same number but not later

**Review by SlimaneAmar — CHANGES_REQUESTED** [review]

**Review by SlimaneAmar — APPROVED** [review]

**Review by SlimaneAmar — APPROVED** [review]

**Pull request merged**